src/OSPSuite.SimModel/include/SimModel/SimulationTask.h

```diff
@@ -29,6 +29,9 @@ class SimulationTask
 	static void SetValuesBelowAbsTolLevelToZero(double * values, int valuesSize, double absTol);
 
 	static void CheckForNegativeValues(Species ** odeVariables, int numberOfVariables, double absTol);
+
+	//mark all parameters which are used in any ODE varaible or observer as unused
+	static void MarkUsedParameters(Simulation * sim);
```

Comment:
That's the main routine, which is called between Simulation.Load and Simulation.Finalize.
It sets `UsedInSimulation ` flag for every simulation parameter